dace/transformation/pass_pipeline.py

```diff
@@ -272,12 +322,18 @@ class Pipeline(Pass):
         print('Promoted scalars:', results['ScalarToSymbolPromotion'])
 
     """
-    passes: List[Pass]
+
+    _passes = properties.ListProperty(element_type=Pass,
+                                      default=[],
+                                      desc='List of passes that this pipeline contains')
+    _pass_names = properties.SetProperty(element_type=str,
+                                         default=set(),
+                                         desc='List of passe names that this pipeline contains')
 
     def __init__(self, passes: List[Pass]):
-        self.passes = []
-        self.pass_names = set(type(p).__name__ for p in passes)
-        self.passes.extend(passes)
+        self._passes = []
+        self._pass_names = set(type(p).__name__ for p in passes)
```

Comment:
why make this a property? it is derived from passes

Reply:
If it does not have a leading underscore it will have to be a property, otherwise the `properties` system complains.

Reply:
not the same issue. Why is _pass_names a property? Should not be.